src/file-events/cpp/generic_fsnotifier.cpp

```diff
@@ -121,14 +127,15 @@ bool AbstractServer::unregisterPaths(const vector<u16string>& paths) {
     return success;
 }
 
-void AbstractServer::terminate() {
+void AbstractServer::terminate(JNIEnv* env) {
     unique_lock<mutex> terminationLock(terminationMutex);
     terminateRunLoop();
     // TODO Parametrize this
     auto status = terminated.wait_for(terminationLock, THREAD_TIMEOUT);
     if (status == cv_status::timeout) {
         throw FileWatcherException("Termination timed out");
```

Comment:
If termination times out, we don't report termination?

Reply:
Mhm, I guess we could report that it was incomplete.